### PR TITLE
[IMP][14.0] maintenance: add hook method for inherit

### PR DESCRIPTION
**Pull request by tranngocson1996**
This PR
-------
When you need to add certain details to the data in order to make a maintenance request. I can't do it using the old code.
As a result, I've included a hook method to be more flexible with customers.
I hope you can pay attention to it. I really need it for my work.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by tranngocson1996** [comment]
@dht-odoo @tde-banana-odoo Can you help me take a look?
Thank you bro !!!

**Comment by tranngocson1996** [comment]
@qdp-odoo can you help me ?
I think it is very simple and does not affect the system. I really need it.
cc: @dht-odoo @tde-banana-odoo

**Comment by tde-banana-odoo** [comment]
Hello,

I am not in charge of maintenance. Maybe @amoyaux  ?

**Comment by amoyaux** [comment]
Hello. We accept the hook only for master version.

**Comment by Yenthe666** [comment]
@amoyaux with all due respect but that is nonsense. The function is not cleanly inheritable. Besides of this there is no signature or logic change so it cannot break signatures.
I also do these kind of PR's from time to time and honestly they're a legit fix :)

**Comment by tranngocson1996** [comment]
Thank you everyone.
I really need this PR. You can merge it. :)
Cc: @Yenthe666 @amoyaux

**Comment by tranngocson1996** [comment]
@Yenthe666 @amoyaux don't forget me ? Plz :(

**Comment by Yenthe666** [comment]
It's not my call. Give Arnold some time to get back to you.

**Comment by amoyaux** [comment]
robodoo r+

**Comment by Yenthe666** [comment]
Thank you for reconsidering it 👍

**Comment by tranngocson1996** [comment]
Thank you bro

File: addons/maintenance/models/maintenance.py [hunk 1]
```diff
@@ -234,7 +234,13 @@ def _create_new_request(self, date):
...
             'duration': self.maintenance_duration,
             'company_id': self.company_id.id or self.env.company.id
-            })
+        }
+
+    def _create_new_request(self, date):
+        self.ensure_one()
+        vals = self._prepare_maintenance_request_vals(date)
+        maintenance_requests = self.env['maintenance.request'].create(vals)
+        return maintenance_requests
```
Comment (royleviindoo) [inline addons/maintenance/models/maintenance.py:243]: @tranngocson1996 this looks better
```suggestion
        return self.env['maintenance.request'].create(vals)
```

Comment (tranngocson1996) [reply addons/maintenance/models/maintenance.py:243]: > @tranngocson1996 this looks better

@royleviindoo thank you bro :D

**Pull request closed without merging**